webrtc.html

```diff
@@ -2582,6 +2636,110 @@
       </dl>
     </section>
 
+    <section>
+      <h2 id="sec.cert-mgmt">Certificate Management</h2>
+      <p>
+        The certificates that <code>RTCPeerConnection</code> instances use to
+        authenticate with peers use the <a><code>RTCCertificate</code></a>
+        interface.  These objects can be explicitly generated by applications
+        using <code>RTCPeerConnection.generateCertificate</code> and provided in
+        the <a><code>RTCConfiguration</code></a> when constructing a
+        new <code>RTCPeerConnection</code> instance.
+      </p>
+      <p>
+        The explicit certificate management functions provided here are
+        optional.  If an application does not provide
+        the <code>certificates</code> configuration option when constructing
+        an <code>RTCPeerConnection</code> a new set of certificates MUST be
+        generated by the <a>user agent</a>.  That set MUST include an ECDSA
+        certificate with a private key on the P-256 curve and a signature with a
+        SHA-256 hash.
+      </p>
+
+      <dl class="idl" title="partial interface RTCPeerConnection">
+        <dt>static Promise&lt;RTCCertificate&gt; generateCertificate
+          (AlgorithmIdentifier keygenAlgorithm)</dt>
+        <dd>
+          <p>This function takes an argument that describes how the private key associated with the certificate is generated.  The <var>keygenAlgorithm</var> argument uses the WebCrypto
+          [[!WebCryptoAPI]] <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#dfn-AlgorithmIdentifier">AlgorithmIdentifier</a>
+          type.  The <var>keygenAlgorithm</var> value MUST be a valid argument
+            to <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#SubtleCrypto-method-generateKey"><code>Crypto.subtle.generateKey</code></a>.</p>
```

Comment:
> I think that you are incorrectly assuming that this needs to use the WebCrypto infrastructure in any way.
> `AlgorithmIdentifier` is just a dictionary. This is just using the already-defined syntax in WebCrypto to describe what it
> means.

Then I think this shares the same fundamental flaw of your previous PR, in which it looks like a duck, but it neither walks, talks, acts like a duck, but you're still calling it a duck.

> Also, we aren't allowing applications to say anything about the signing process. You're assuming a lot when you 
> invent step 2 there.
>
> I find the W3C habit of overspecificity in specs stifling. What is wrong with describing inputs and outputs?

I'm surprised you didn't realize how closely related these two statements are. The fact that reasonable people (or at least, I hope I am), the people most likely to be implementing or involved it in, the people responsible for this, can reach different conclusions from what _you_ meant is **exactly** why the "stifling" process of specing exists. To ensure that we can get compatibility.

You said the operation must support signing. I described to you how, to Web Crypto, one describes whether or not the algorithm supports signing. That's "clearly" not what you meant, but that's not at all clear in what you propose.

The habit is not saying "You must do exactly X" - it's that you must, from the observer's point of view, have taken these inputs and produced these outputs. Any side-effects of the outputs that can be observed by script **MUST** be consistent between implementations, as decades of experience have shown us people _will_ come to rely on this.

> To that end, I will definitely reference the normalization algorithm, though it's really only a hint for implementations.

I don't really care what you do, and it was more an example of how you _might_ do it, one that was admittedly flawed. What I want to ensure with this PR is that it's unambiguous what needs to happen, and as it stands, it's quite ambiguous and (as a result), not _consistently_ implementable.